frontend/src/components/CompanyOffices/index.js

```diff
@@ -0,0 +1,207 @@
+import React, {Component} from 'react';
+import {connect} from 'react-redux'
+import {addCompanyForOffice, addNewCompany, fetchOfficesByCompany, setNewHeadquarter} from "../../API";
+import Company from "../Company";
+import './index.css'
+import Input from "@material-ui/core/Input";
+import Box from "@material-ui/core/Box";
+import InputLabel from "@material-ui/core/InputLabel";
+import Typography from "@material-ui/core/Typography";
+import TableHead from "@material-ui/core/TableHead";
+import TableRow from "@material-ui/core/TableRow";
+import TableCell from "@material-ui/core/TableCell";
+import TableBody from "@material-ui/core/TableBody";
+import Table from "@material-ui/core/Table";
+import Button from "@material-ui/core/Button";
+
+
+const mapStateToProps = (state, ownProps) => {
+    return {
+        company: state.companies.filter((company) => {
+            return company.id === parseInt(ownProps.match.params.id)
+        })
+    }
+};
+
+class CompanyOffices extends Component {
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            officesData: null,
+            showAddForm: false,
+            formData: {
+                city: '',
+                country: '',
+                street: '',
+                postalCode: '',
+                monthlyRent: '',
+            }
+        }
+    }
+
+
+    onInputChange(e) {
+        const {id, value} = e.target;
+        this.setState({
+            ...this.state,
+            formData: {
+                ...this.state.formData,
+                [id]: value
+            }
+        })
+    }
+
+    handleSubmit(e) {
+
+        const {formData} = this.state;
+        addCompanyForOffice(formData, this.props.company[0].id)
+            .then((response) => {
+                console.log(response)
+            })
+            .catch((error) => {
+                console.log(error)
+            })
+    }
+
+
+    componentDidMount() {
+        fetchOfficesByCompany(this.props.match.params.id)
+            .then((response) => {
+                this.setState({
+                    officesData: response.data
+                })
+            }).catch((error) => {
+            console.log(error)
+        })
+    }
+
+
+    setHeadquarter(id) {
+        const company_id = this.props.company[0].id;
+        setNewHeadquarter(id, this.props.company[0].id, this.props.company[0].name)
```

Comment:
same here

Reply:
Yes, I've used the same approach here too

Reply:
Yes, I've used the same aproach here too

Reply:
I meant we need to handle response of response on other where, outside of component

Reply:
Sorry, but I didn't actually get that do you mean

Reply:
ok, its fine